nimbus-core/src/main/java/com/antheminc/oss/nimbus/domain/model/state/repo/AbstractWSModelRepository.java

```diff
@@ -0,0 +1,128 @@
+		this.restTemplate = beanResolver.get(RestTemplate.class);
+	}
+
+	abstract public String getTargetUrl(String alias);
+	
+	abstract public String constructSearchUri(SearchCriteria<?> searchCriteria);
+	
+	abstract public <ID extends Serializable, T> T handleGet(ID id, Class<T> referredClass, URI url);
+	
+	abstract public <T> Object handleSearch(Class<T> referredDomainClass, Supplier<SearchCriteria<?>> criteriaSupplier, String alias, URI uri);
+	
+	protected URI createUriForAlias(String alias, String url) {
+		String targetUrl = getTargetUrl(alias);
+		if(targetUrl == null)
+			return null;
+		
+		String urlToConstruct = StringUtils.startsWith(url, "/") ? url : "/".concat(url);
+		urlToConstruct = targetUrl.concat(urlToConstruct);
+		try {
+			URI uri = new URI(urlToConstruct);
+			return uri;
+		} catch (URISyntaxException e) {
+			throw new FrameworkRuntimeException("Cannot create URI from supplied url: "+url);
+		}
+	}
+	
+	public <ID extends Serializable, T> T _get(ID id, Class<T> referredClass, String alias, String url) {
+		URI uri = createUriForAlias(alias, url);
+		if(uri == null)
+			return null;
+
+		try {
+			return handleGet(id, referredClass, uri);
+		} catch(Exception e) {
+			logit.error(() -> "Exception occured while calling Web service "+e.getMessage());
+			throw new FrameworkRuntimeException(e.getMessage(), e);
+		}
+	}
+	
+	public <T> Object _search(Class<T> referredDomainClass, String alias, Supplier<SearchCriteria<?>> criteriaSupplier) {
+		SearchCriteria<?> searchCriteria = criteriaSupplier.get();
+		String searchUri = constructSearchUri(searchCriteria);
+		URI uri = createUriForAlias(alias, searchUri);
+		if(uri == null)
+			return null;
+		
+		return handleSearch(referredDomainClass, criteriaSupplier, alias, uri);
+	}
+	
+	protected Object execute(Supplier<RequestEntity<?>> reqEntitySupplier, Supplier<ParameterizedTypeReference<?>> responseTypeSupplier) {
+		 try {
+			ResponseEntity<?> responseEntity = getRestTemplate().exchange(reqEntitySupplier.get(), responseTypeSupplier.get());
+			return Optional.of(responseEntity).map((response)->response.getBody()).orElse(null);
+		} catch(Exception e) {
+			handleException(e, reqEntitySupplier.get().getUrl());
+		}			 
+		 return null;
+	}
+	
+	public class CustomParameterizedTypeImpl implements ParameterizedType {
+	    private ParameterizedType delegate;
+	    private Type[] actualTypeArguments;
+
+	    public CustomParameterizedTypeImpl(ParameterizedType delegate, Type[] actualTypeArguments) {
+	        this.delegate = delegate;
+	        this.actualTypeArguments = actualTypeArguments;
+	    }
+
+	    @Override
+	    public Type[] getActualTypeArguments() {
+	        return actualTypeArguments;
+	    }
+
+	    @Override
+	    public Type getRawType() {
+	        return delegate.getRawType();
+	    }
+
+	    @Override
+	    public Type getOwnerType() {
+	        return delegate.getOwnerType();
+	    }
+	}
+	
+	private void handleException(Exception e, URI uri) {
```

Comment:
Similar comments provided earlier relating to logging an ex and then immediately turning around and throwing a new wrapped exception.